cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringProvisioner.java

```diff
@@ -68,6 +77,19 @@ public ProvisionerSpecification getSpec() {
   public void validateProperties(Map<String, String> properties) {
     // Creates the TetheringConf for validation
     TetheringConf.fromProperties(properties);
+    String peer = properties.get(TetheringConf.TETHERED_INSTANCE_PROPERTY);
+    String namespace = properties.get(TetheringConf.TETHERED_NAMESPACE_PROPERTY);
+    PeerInfo peerInfo;
+    try {
+      peerInfo = tetheringStore.getPeer(peer);
+    } catch (Exception e) {
+      throw new IllegalArgumentException(String.format("%s is not a tethered peer", peer));
```

Comment:
Exception is pretty broad, it can fail for a lot of reasons. It's better to catch the PeerNotFoundException and rethrow with this message, and catch the IOException and rethrow with some other error message.

Reply:
done